src/config.rs

```diff
@@ -0,0 +1,121 @@
+use serde::Deserialize;
+
+#[derive(Debug, Clone)]
+pub struct Remote {
+    pub name: String,
```

Comment:
I'm not really sure what `name` is supposed to be in relation to a remote host. It's also never read. Is it just like a comment field in the config file?

Reply:
It isn't a problem as the `name` is optional and it will pick the first `remote` if no `remote-name` is specified in the cli. See line 90 in `config.rs`.

Reply:
Yeah, but still, why would you want it? Is there a use case where an ssh config entry isn't the better choice? Rebuilding existing functionality poorly isn't good imo and just adds complexity.

Reply:
Ok, I didn't knew the ssh entry. But then, what's the point of having a config file? Further the split of host & user is counterproductive.

Reply:
> But then, what's the point of having a config file?

The ssh config file won't tell `cargo-remote` *which* server to use for building. The `cargo-remote` config file is still useful to avoid re-typing the host every time (assuming you use the same every time).

> the split of host & user is counterproductive

You are right, I hadn't thought of that really :laughing:

Reply:
... and the name specified in the ssh-config can be used with the cli argument. But then, what's the point of a separate port in the config?

Reply:
> But then, what's the point of a separate port in the config?

I [pointed that out](https://github.com/sgeisler/cargo-remote/pull/13#issuecomment-634534941) in #13, but at least as an option it makes _some_ sense: you don't want to write an ssh config entry e.g. for an one-off test. It's not really needed in the config unless one wants to mirror all options (which might be useful for ease of documentation: _"you can use every option as a config entry too"_).

Reply:
As I started the reverse process, I stumbled over the `temp_dir` Option which definitely can't be exported to the ssh-config and it could be quite useful for more complex scenarios.

Reply:
That might be a reason to have different host configs in our config file. Do you consider having different build hosts with different build directories as common enough to add the complexity? I feel like in that case you can just as well specify in in the options (I view it as a _very_ rare corner case). But if you think it's common (or concretely useful to you) I'd be ok with it.

Reply:
I would use it on my laptop depending on where I am. Another use case I can see is in companies with large projects which would take forever on underpowered clients and there administrators, as far as i know, want to control every aspect of such a software through a config-file. But you are the maintainer and if it's too complex for a small use-case I will discard it.